hedera-mirror-rosetta/app/services/account_service_test.go

```diff
@@ -0,0 +1,183 @@
+
+	assert.IsType(t, &AccountAPIService{}, accountService)
+}
+
+func TestAccountBalance(t *testing.T) {
+	// given:
+	exampleBlock := &types.Block{
+		Index:               1,
+		Hash:                "0x123jsjs",
+		ConsensusStartNanos: 1000000,
+		ConsensusEndNanos:   20000000,
+		ParentIndex:         2,
+		ParentHash:          "0xparenthash",
+	}
+
+	expectedAccountBalanceResponse := &rTypes.AccountBalanceResponse{
+		BlockIdentifier: &rTypes.BlockIdentifier{
+			Index: 1,
+			Hash:  "0x123jsjs",
+		},
+		Balances: []*rTypes.Amount{
+			{
+				Value:    "1000",
+				Currency: config.CurrencyHbar,
+			},
+		},
+	}
+
+	exampleAmount := &types.Amount{
+		Value: int64(1000),
+	}
+
+	mocks.Setup()
+	mocks.MBlockRepository.On("RetrieveLatest").Return(exampleBlock, mocks.NilError)
+	mocks.MAccountRepository.On("RetrieveBalanceAtBlock").Return(exampleAmount, mocks.NilError)
+
+	commons := NewCommons(mocks.MBlockRepository, mocks.MTransactionRepository)
+	accountService := NewAccountAPIService(commons, mocks.MAccountRepository)
+
+	// when:
+	actualResult, e := accountService.AccountBalance(nil, exampleRequest(false))
+
+	// then:
+	assert.Equal(t, expectedAccountBalanceResponse, actualResult)
+	assert.Nil(t, e)
+}
+
+func TestAccountBalanceFullData(t *testing.T) {
+	// given:
+	exampleBlock := &types.Block{
+		Index:               1,
+		Hash:                "0x123jsjs",
+		ConsensusStartNanos: 1000000,
+		ConsensusEndNanos:   20000000,
+		ParentIndex:         2,
+		ParentHash:          "0xparenthash",
+	}
+
+	expectedAccountBalanceResponse := &rTypes.AccountBalanceResponse{
+		BlockIdentifier: &rTypes.BlockIdentifier{
+			Index: 1,
+			Hash:  "0x123jsjs",
+		},
+		Balances: []*rTypes.Amount{
+			{
+				Value:    "1000",
+				Currency: config.CurrencyHbar,
+			},
+		},
+	}
+
+	exampleAmount := &types.Amount{
+		Value: int64(1000),
+	}
+
+	mocks.Setup()
+	mocks.MBlockRepository.On("FindByIdentifier").Return(exampleBlock, mocks.NilError)
+	mocks.MAccountRepository.On("RetrieveBalanceAtBlock").Return(exampleAmount, mocks.NilError)
+
+	commons := NewCommons(mocks.MBlockRepository, mocks.MTransactionRepository)
+	accountService := NewAccountAPIService(commons, mocks.MAccountRepository)
+
+	// when:
+	actualResult, e := accountService.AccountBalance(nil, exampleRequest(true))
+
+	//then:
+	assert.Equal(t, expectedAccountBalanceResponse, actualResult)
+	assert.Nil(t, e)
+	mocks.MBlockRepository.AssertNotCalled(t, "RetrieveLatest")
+}
+
+func TestAccountBalanceThrows(t *testing.T) {
+	// given:
+	mocks.Setup()
+	mocks.MBlockRepository.On("RetrieveLatest").Return(mocks.NilBlock, &rTypes.Error{})
+
+	commons := NewCommons(mocks.MBlockRepository, mocks.MTransactionRepository)
+	accountService := NewAccountAPIService(commons, mocks.MAccountRepository)
+
+	// when:
+	actualResult, e := accountService.AccountBalance(nil, exampleRequest(false))
+
+	// then:
+	assert.Nil(t, actualResult)
+	assert.NotNil(t, e)
```

Comment:
I think that it would be better to check that the error type is correct, not only that its not nill